contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,190 @@
+import { HEROES } from './mock-heroes';
+@Injectable({
+  providedIn: 'root',
+})
+export class HeroService {
+  getHeroes() { return HEROES; }
+}
+```
+The `HeroesComponent` declared previously, can make use of this `HeroService` by injecting the dependency in it's constructor as a parameter.
+
+```
+constructor(heroService: HeroService) {
+ this.heroes = heroService.getHeroes();
+}
+```
+
+In the above example, we write the `getHeroes` method only once in the service class and we can access this method from any component by injecting the service in the constructor.
+This explains how DI increases "efficiency and modularity".
+
+Declarative Templates and Dependency Injection are only a couple of noteworthy features of Angular. A more comprehensive [list](https://angular.io/features) is available on the official website. 
+ 
+## Why Angular?
+
+Now that we know what Angular is, let us look at some benefits it has to offer.
+
+### Benefit 1: Angular provides Good Tooling Services.
+
+Angular provides developers with tools that enforce "best practices". Let us look at some of these tools.
+
+* **TypeScript** - Angular applications are built using the TypeScript language, a superset of JavaScript.
+TypeScript provides <tooltip content="Static typing involves type checking during compile-time as opposed to run-time. Types are associated with variables not values.">static typing</tooltip>.
+This helps the compiler show warnings about any potential errors in the code, before the app runs. Consider the following example: 
+
+    ```typescript
+    function add(x : number, y : number) : number {
+        return x + y;
+    }
+    add('a', 'b'); // compiler flags an error
+    ```
+    In the code above, parameters `x` and `y` are declared to have the type `number`. Thus, the compiler shows an error when the function is called with strings as inputs to the function.
+    Another advantage is that TypeScript provides code completion using <tooltip content="IntelliSense provides active hints as code is added">IntelliSense</tooltip>.
+    
+* **Angular CLI** - The [official documentation](https://angular.io/cli) claims that the Command-line Interface can be used to develop and maintain applications. It can be used to enforce "best practices" amongst developers.
+    
+    The Angular CLI has out-of-the-box integration with <tooltip content="Codelyzer is a set of rules for static code analysis of Angular projects">**Codelyzer**</tooltip>.
+    Codelyzer helps developers write high quality code by linting it against the [official Angular style guide](https://angular.io/guide/styleguide).
+        
+    With the CLI we can also perform the following commands to ease the development process. You can refer to the official documentation for a complete list of the CLI commands.
+     - Generate a new Angular application with - `ng new`
+     - Generate Angular files - `ng generate`
+     - Build our application for deployment - `ng build`
```

Comment:
No need for this three here.  If at all, they should be given as examples when you mentioned that Angular CLI can manage web apps.